book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+| PipelineRun | Post Final PipelineRun | Check the result of the Final PipelineRun, check ReleaseLink and create a Release |
+
+### Pipeline
+
+The test pipeline will consist of a single definition that could be executed as two logically distinct pipeline runs by the Integration Service. The logical pipeline runs are:
+
+- Component (Prelim) pipeline run
+- Composite (Final) pipeline run
+
+The Component pipeline run is always triggered for each Component of an Application and is triggered by a completed build of a component.  
+
+The Composite run is only triggered when it has been determined by the Integration Service that two or more Components of an Application have been successfully built concurrently by their respective Component pipeline and need to be tested together. 
+
+The two stage model is roughly modeled on the 'fast loop / slow loop' model that OpenShift uses.  ART is the slow loop.  The global candidates list at some point in time is the set of artifacts that the Composite pipeline will test and mark as a 'shippable' snapshot.
+
+[Draft: StoneSoup-4 Pipeline & Task Design](https://docs.google.com/document/d/1ucQI3WJzkLpueYLuzHEXpKNvYQafDl4gNBZryvi2d5o/edit#)
+
+### PipelineRun
+
+The Integration Service will generate a PipelineRun that can reference the user provided tekton pipeline. 
+
+#### Pipeline Reference
+The pipeline reference should point to a well known location that an Application team can provide their tekton bundle. The pipeline packaging will be retrieved the IntegrationScenario CR
+
+#### Annotations/Labels
+
+Following the [guidelines](https://docs.google.com/document/d/1gyXM3pkKFFfxHZnopBi_53vREFWhwA0pFUuIhopDuEo/edit#) in we can define some basic ones to start
+```
+“pipelines.appstudio.openshift.io/type": "test",
+"test.appstudio.openshift.io/test":      “<component|composite>”
+“test.appstudio.openshift.io/scenario":  “<IntegrationTestScenario name>”
+"appstudio.openshift.io/snapshot":       “<snapshot name>”
+“appstudio.openshift.io/component":      “<component name>”
+“appstudio.openshift.io/application":    “<Application name>”
+```
+We must also copy the annotations and labels from the Build PipelineRun and append those to the Test PipelineRuns for traceability across the system per [Labels and Annotations for StoneSoup pipelines](https://docs.google.com/document/d/1fJq4LDakLfcAPvOOoxxZNWJ_cuQ1ew9jfBjWa-fEGLE/edit#) and [StoneSoup builds and tests PRs](https://docs.google.com/document/d/113XTplEWRM63aIzk7WwgLruUBu2O7xVy-Zd_U6yjYr0/edit#)
+
+#### Workspace
+#### Persistent Volume
+ A persistent volume that the pipeline can store test artifacts generated from the test execution.
+
+#### Secrets
+
+We may need secrets mounted so that the `Environment Provisioner` Task running in the workspace has the correct permissions to create AppStudio Application Environments resources within the Applications/Component’s namespace
```

Comment:
We can change `We may need secrets mounted` to `The Integration service needs secrets mounted`.